container/kvm/run_linux.go

```diff
@@ -14,17 +15,25 @@ import (
 	"github.com/juju/errors"
 )
 
-// run the command as user libvirt-qemu and return the combined output.
-func runAsLibvirt(command string, args ...string) (string, error) {
+// Run the command as user libvirt-qemu and return the combined output.
+// If dir is non-empty, use it as the working directory.
+func runAsLibvirt(dir, command string, args ...string) (string, error) {
 	uid, gid, err := getUserUIDGID(libvirtUser)
 	if err != nil {
 		return "", errors.Trace(err)
 	}
 
-	logger.Debugf("running: %s %v", command, args)
+	cmd := exec.Command(command, args...)
+	if dir != "" {
+		cmd.Dir = dir
+	}
+
+	if dir == "" {
+		dir, _ = os.Getwd()
```

Comment:
This is much cleaner now, so we don't have to push and pop in a filesystem, which as you say, was never going to be gorountine safe.